compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -264,30 +178,66 @@ private void validateRecordFieldsAnnotation(SyntaxNodeAnalysisContext ctx, Node
                     validateReadOnly(ctx, fieldNode.readonlyKeyword().isPresent(), fieldNode.location());
                 }
                 Optional<MetadataNode> metadataNode = fieldNode.metadata();
-                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.typeName().toSourceCode().trim(),
+                metadataNode.ifPresent(node -> validateAnnotationFields(ctx, node,
+                        fieldNode.typeName().toSourceCode().trim(),
                         fieldNode.location(), memberNodes, fieldNode.fieldName().text().trim()));
             } else if (field instanceof RecordFieldWithDefaultValueNode) {
                 RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) field;
                 if (primaryKeys.contains(fieldNode.fieldName().text().trim())) {
                     validateReadOnly(ctx, fieldNode.readonlyKeyword().isPresent(), fieldNode.location());
                 }
                 Optional<MetadataNode> metadataNode = fieldNode.metadata();
-                metadataNode.ifPresent(node -> validateAnnotation(ctx, node, fieldNode.typeName().toSourceCode().trim(),
+                metadataNode.ifPresent(node -> validateAnnotationFields(ctx, node,
+                        fieldNode.typeName().toSourceCode().trim(),
                         fieldNode.location(), memberNodes, fieldNode.fieldName().text().trim()));
             }
         }
     }
 
+    private void checkRecordFieldsAnnotation(SyntaxNodeAnalysisContext ctx, Node recordNode) {
```

Comment:
This method has nearly the same functionality as `validateRecordFieldsAnnotation()`?

Reply:
Updated to `checkFieldAnnotationPresent`